src/test/java/org/fenixedu/academic/domain/StranglerFigSwitchBibliographicReferencesTest.java

```diff
@@ -0,0 +1,122 @@
+        //ADD
+        final CompetenceCourseInformation courseInformation =
+                competenceCourseA.getCompetenceCourseInformationsSet().iterator().next();
+        assertTrue(courseInformation.getBibliographicReferences().getBibliographicReferencesList().isEmpty());
+        assertTrue(courseInformation.getBibliographiesSet().isEmpty());
+
+        BibliographicReferences bibliographicReferences = courseInformation.getBibliographicReferences();
+        bibliographicReferences = bibliographicReferences.with("2024", "Felicidade na Produtividade de Trabalho",
+                "Sofia Nascimento", "qubIT", null, BibliographicReferenceType.MAIN);
+        courseInformation.setBibliographicReferences(bibliographicReferences);
+
+        courseInformation.getBibliographiesSet().stream().forEach(br -> br.delete());
+        courseInformation.getBibliographiesSet()
+                .addAll(BibliographicReference.createFromBibliographicReferences(bibliographicReferences));
+
+        assertTrue(courseInformation.getBibliographicReferences().getBibliographicReferencesList().size() == 1);
+        assertTrue(courseInformation.getBibliographiesSet().size() == 1);
+        assertTrue(equals(courseInformation.getBibliographicReferences().getBibliographicReferencesList().get(0),
+                courseInformation.getBibliographiesSet().iterator().next()));
+
+        //"EDIT"
+        bibliographicReferences = courseInformation.getBibliographicReferences().replacing(0, "2024",
+                "Como a Qualidade do código cria Bem-Estar", "Sofia Nascimento", "qubIT", null, BibliographicReferenceType.MAIN);
+        courseInformation.setBibliographicReferences(bibliographicReferences);
+
+        courseInformation.getBibliographiesSet().stream().forEach(br -> br.delete());
+        courseInformation.getBibliographiesSet()
+                .addAll(BibliographicReference.createFromBibliographicReferences(bibliographicReferences));
+
+        assertTrue(courseInformation.getBibliographicReferences().getBibliographicReferencesList().size() == 1);
+        assertTrue(courseInformation.getBibliographiesSet().size() == 1);
+        assertTrue(equals(courseInformation.getBibliographicReferences().getBibliographicReferencesList().get(0),
+                courseInformation.getBibliographiesSet().iterator().next()));
+
+        //DELETE
+        bibliographicReferences = courseInformation.getBibliographicReferences().without(0);
+        courseInformation.setBibliographicReferences(bibliographicReferences);
+
+        courseInformation.getBibliographiesSet().stream().forEach(br -> br.delete());
+        courseInformation.getBibliographiesSet()
+                .addAll(BibliographicReference.createFromBibliographicReferences(bibliographicReferences));
+
+        assertTrue(courseInformation.getBibliographicReferences().getBibliographicReferencesList().isEmpty());
+        assertTrue(courseInformation.getBibliographiesSet().isEmpty());
+    }
```

Comment:
Is it possible to add some tests with multiple ordered references (and not just for one single entry) ?